apps/kol-amelamdim/pages/category/[category].tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  useEffect,
+  useState,
+  ChangeEvent,
+  useContext,
+  ReactElement,
+} from 'react';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  TablePagination,
+  Link,
+  Button,
+} from '@mui/material';
+import { useRouter } from 'next/router';
+import { Category, IFile } from '@kol-amelamdim/types';
+import { StyledPageContainer } from '@kol-amelamdim/styled';
+import { FILE_TYPES_DICTIONARY } from '@kol-amelamdim/constants';
+import { API_ERRORS } from '@kol-amelamdim/api-errors';
+import { FilterCard } from '../../components/filter-card/FilterCard';
+import { useCategoriesFiles } from '../../hooks/use-categories-files';
+import { AuthContext } from '../../context/auth-context-provider';
+import { UploadFileDialog } from '../../components';
+import { AlertContext } from '../../context/alert-context-provider';
+import { AlertLayout } from '../../layouts';
+
+const Mivhanim = () => {
+  const [fileType, setFileType] = useState('');
+  const [filterText, setFilterText] = useState('');
+  const [page, setPage] = useState<number>(0);
+  const [rowsPerPage, setRowsPerPage] = useState<number>(5);
+  const [filteredFiles, setFilteredFiles] = useState<IFile[]>([]);
+  const [isUploadFileDialogOpen, setIsUploadFileDialogOpen] = useState(false);
+
+  const router = useRouter();
+  const { category } = router.query;
+  const { files, error, loading } = useCategoriesFiles(category as Category);
+  const { isAuthenticated } = useContext(AuthContext);
+  const { setAlertMessage, setAlertType } = useContext(AlertContext);
+
+  //TODO: handle Error & loading & no data to show.
+  //TODO: handle file upload to be added to the tabled immediately
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const handleFilter = () => {
+    let filteredFileType = [...files];
+    if (
+      fileType &&
+      FILE_TYPES_DICTIONARY[fileType] !== FILE_TYPES_DICTIONARY[10]
```

Comment:
I would extract this condition to variable with meaningful name beause it took me a while to understand it

Reply:
Resolved.